nixos/modules/services/network-filesystems/openafs/server.nix

```diff
@@ -101,6 +163,39 @@ in {
         description = "Definition of all cell-local database server machines.";
       };
 
+      authentication = {
+        principal = mkOption {
+          default = "afs/${cfg.cellName}@${lib.toUpper cfg.cellName}";
```

Comment:
The Kerberos realm should be the configured default realm `security.krb5.settings.libdefaults.default_realm` and not assume that the AFS cell runs in its own realm.

Reply:
Thank you, I also took the liberty of redefining the option definition to be more fool-proof in case of refactoring of the Kerberos module